netcreateorg/netcreate-2018
# Filter Feature

This implements #167.

Branch: `dev-bl/filter`

This introduces "Highlighting" and "Filtering" as separate but related features.

Whimsical [graph data flow diagram](https://whimsical.com/d3-data-flow-B2tTGnQYPSNviUhsPL64Dz)

# To Test

1. `git fetch; git checkout dev-bl/filter`
2. `./nc.js --dataset=<yourdataset` -- Open a dataset that has a number of nodes and edges.
3. Click "Highlight/Filter" tab
4. Select Label: "contains".
5. Type in a seach value, e.g. "the"
6. The system should fade away any nodes that do not contain "the" -- note edges should fade too
7. Click the "Filter" tab.
8. The system should redraw, this time removing any nodes that do not contain "the".

The idea here is that the highlight/filter form acts the same across both conditions.  This way you can easily toggle back and forth between fading (highlighting) and removing (filtering).

9. Click the "Graph" tab.
10. You should see a summary of the current highlight/filter setting in a blue bar under "Graph".  It should indicate whether you have highlight or filter active.
11. You should be able to add and edit nodes and edges.

NOTE if you add a new node that is filtered out, it won't display.

---

As noted in #167, there are still a number of related features that are outstanding.
